[PyTorch] Make RecordFunction::active private

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47550 [PyTorch] Lazily construct guts of RecordFunction
* **#47549 [PyTorch] Make RecordFunction::active private**

In preparation for moving state onto the heap.

Differential Revision: [D24812214](https://our.internmc.facebook.com/intern/diff/D24812214/)

swolchok committed:
[PyTorch] Make RecordFunction::active private

In preparation for moving state onto the heap.

Differential Revision: [D24812214](https://our.internmc.facebook.com/intern/diff/D24812214/)

[ghstack-poisoned]

swolchok committed:
s/TORCH_CHECK/TORCH_INTERNAL_ASSERT_DEBUG_ONLY on "[PyTorch] Make RecordFunction::active private"


In preparation for moving state onto the heap.

Differential Revision: [D24812214](https://our.internmc.facebook.com/intern/diff/D24812214/)

[ghstack-poisoned]

swolchok committed:
revert stray TORCH_CHECK removal on "[PyTorch] Make RecordFunction::active private"


In preparation for moving state onto the heap.

Differential Revision: [D24812214](https://our.internmc.facebook.com/intern/diff/D24812214/)

[ghstack-poisoned]

ilia-cher commented on aten/src/ATen/record_function.h:
```diff
@@ -193,12 +193,17 @@ struct TORCH_API RecordFunction {
 
-  // Whether this RecordFunction runs any callbacks
-  bool active = false;
+  bool isActive() const {
```
mark it as inline too?

swolchok committed:
Update on "[PyTorch] Make RecordFunction::active private"


In preparation for moving state onto the heap.

Differential Revision: [D24812214](https://our.internmc.facebook.com/intern/diff/D24812214/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Make RecordFunction::active private"


In preparation for moving state onto the heap.

Differential Revision: [D24812214](https://our.internmc.facebook.com/intern/diff/D24812214/)

[ghstack-poisoned]
